rust/private/rust.bzl

```diff
@@ -647,7 +648,7 @@ _common_attrs = {
             List of `rust_proc_macro` targets used to help build this library target.
         """),
         cfg = "exec",
-        providers = [rust_common.crate_info],
+        providers = [[rust_common.crate_group_info], [rust_common.crate_info]],
```

Comment:
This accepts `crate_group_info` providers, but then doesn't look for them? I think you'd need a second branch in the `if` you added which looks for `crate_group_info` and if it's present, iterates across them to look in their `crate_info` values?

Please could you add a test which uses this functionality, so we can see the code definitely works?

Reply:
> This accepts crate_group_info providers, but then doesn't look for them? I think you'd need a second branch in the if you added which looks for crate_group_info and if it's present, iterates across them to look in their crate_info values?

This follows the established checking logic for deps, on top of this change. So I didn't add a second branch. Also, as `crate_group_info` returns a depset, do we want to flatten that just for checking?